### PR TITLE
Docker GPU

### PR DESCRIPTION
**Pull request by hhsecond**
Fixing #120

### CODE REVIEW

**Comment by hhsecond** [comment]
Few points to note here
- For making the server accessible outside, We needed `--protected-mode no`. Not sure if there is a better way
- We need to push this to redisai docker hub for using it in the CI. I have pushed one to [tensorwerk's account](https://hub.docker.com/r/tensorwerk/redisai-gpu). Once this PR is good to go, it would be great if somebody who has access could help me pushing this to redisai docker hub

File: get_deps.sh [hunk 1]
```diff
@@ -5,21 +5,6 @@ HERE="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
 set -e
 [[ $VERBOSE == 1 ]] && set -x
 
-if [[ "$1" == "cpu" ]]; then
```
Comment (rafie) [inline get_deps.sh]: Let's not remove this for the time being, until gen_deps.sh is stabilized (there's quite a few changes coming up with the ARM PR).

Comment (hhsecond) [reply get_deps.sh]: Done

**Comment by lantiga** [comment]
Regarding protected mode, I think (but I might be wrong @rafie) the way to go would be to not specify it.
Instead, one could mount redis.conf from the host or a docker volume with the appropriate settings (eg auth, ...).
What do you think?

**Comment by hhsecond** [comment]
I think we should have `protected mode no` by default and let the user override that by mounting a config (I assume this is what base Redis image and RediAI CPU image does). I might have to move `protected mode no` from the command to the config to make it overridable

**Comment by lantiga** [comment]
Thank you Sherin!

**Comment by hhsecond** [comment]
This is what `Redis` does. It removes the config from the source before building `Redis` itself
https://github.com/docker-library/redis/blob/0b2910f292fa6ac32318cb2acc84355b11aa8a7a/5.0/Dockerfile#L66

Instead what I did is to handle that using the config file which can be controlled by mounting the configuration file. Also, the installation of Redis in the old `Dockercompose-GPU` was from ubuntu repo and there were few dependency issues. I had to install it from the `stable` URL. Hope that's OK

**Comment by lantiga** [comment]
Looks good to me. I'm good to merge, @rafie?

**Pull request merged**